src/Caller.sol

```diff
@@ -101,36 +110,47 @@ contract Caller is EIP712("Caller", "1"), ERC2771Context(address(this)) {
     /// @param unauthorized The authorized address.
     event Unauthorized(address indexed sender, address indexed unauthorized);
 
+    /// @notice Emitted when revoking all authorizations to make calls on behalf of the `sender`.
+    /// @param sender The authorizing address.
+    event UnauthorizedAll(address indexed sender);
+
     /// @notice Grants the authorization of an address to make calls on behalf of the sender.
     /// @param user The authorized address.
     function authorize(address user) public {
         address sender = _msgSender();
-        require(_authorized[sender].add(user), "Address already is authorized");
+        require(_getAuthorizedSet(sender).add(user), "Address already is authorized");
         emit Authorized(sender, user);
     }
 
     /// @notice Revokes the authorization of an address to make calls on behalf of the sender.
     /// @param user The unauthorized address.
     function unauthorize(address user) public {
         address sender = _msgSender();
-        require(_authorized[sender].remove(user), "Address is not authorized");
+        require(_getAuthorizedSet(sender).remove(user), "Address is not authorized");
         emit Unauthorized(sender, user);
     }
 
+    /// @notice Revokes all authorizations to make calls on behalf of the sender.
+    function unauthorizeAll() public {
```

Comment:
Each authorized address could `unauthorizeAll` all other authorized addresses. (including itself).

Reply:
A scenario could be that I don't like a specific action another authorized address wants to perform. 
I front-run it with an `unauthorizeAll` call to delay the action.

Reply:
Yes, cleaning all authorizations while being authorized removes your own authorization too, that's probably the expected thing that should happen. You can explicitly add an exception by batching `unauthorizeAll` with a bunch of `authorize`s, e.g. re-authorizing yourself. 

Yes, all of this is correct. It's the same as with `unauthorize`, each authorized address may unauthorize the others, `unauthorizeAll` is just an optimization.

Reply:
Yes, but why do we think is it better than keeping this method only callable by the actual owner?